android/src/main/java/com/rnldk/RnLdkModule.kt

```diff
@@ -346,6 +434,28 @@ class RnLdkModule(private val reactContext: ReactApplicationContext) : ReactCont
     }
   }
 
+  @ReactMethod
+  fun payInvoice(bolt11: String, amtSat: Int, promise: Promise) {
+    if (channel_manager_constructor?.payer == null) return promise.reject("payer is null");
+
+    val parsedInvoice = Invoice.from_str(bolt11)
+    if (parsedInvoice !is Result_InvoiceNoneZ.Result_InvoiceNoneZ_OK) {
+      return promise.reject("cant parse invoice");
+    }
+
+    val sendRes = if (amtSat != 0) {
+      channel_manager_constructor!!.payer!!.pay_zero_value_invoice(parsedInvoice.res, amtSat.toLong() * 1000)
```

Comment:
To use the payer you need to update the `PaymentPathFailed` handling to instead `PaymentFailed`. If no `payer` is set (ie net graph or scorer are null) then you will want to call `abandon_payment` on `PaymentPathFailed` to generate a `PaymentFailed` immediately instead of waiting for a timeout.

Reply:
@TheBlueMatt fixed, can you pls take a look again?